core/jobs.py

```diff
@@ -597,16 +794,37 @@ def __iter__(self):
 
     @classmethod
     def from_json(cls, input_shard_state):
```

Comment:
I looked at the input/output for this and to_json, and it seems to be coming from some configuration data that isn't kept in the code..?  https://pastebin.com/dwxf5NbK shows some sample values I got out. I've grepped for the keys in those dicts and couldn't find them anywhere, so I'm not sure where-else to look!